### DISCUSSION TITLE
Is there a way to generate custom social card per blog post

### DISCUSSION THREAD

**Discussion by jcubic** [facebook/docusaurus#9941] [Q&A]
I have Elevently blog and I use special filter to run the code that generate twitter card based on SVG file, I use puppeteer to generate png file.

Does Docusaurus have a way to hook up into the system and generate the card? The problem is that the card need to be generated before the build otherwise Docusaurus will throw an error it can't find a file. Other problem is that I need access authors and title of the blog post.

Do I need to write my own code to parse Docusaurus files to generate Twitter Cards or is there a way to make it part of the build? I wanted to swizzle BlogPost, but Docusaurus says it's unsafe.

**Comment by slorber** [comment]
We do not support this feature out of the box but we would like for a while now

- https://github.com/facebook/docusaurus/pull/7836
- https://github.com/facebook/docusaurus/issues/2968

But I plan to resume the work on this. My intern @OzakIOne worked on a PR to support social card generation through edge rendering and through a Node.js API (https://github.com/OzakIOne/local-og-image-generator/pull/1). Now we need to figure out how to integrate that and design a flexible API for this, unfortunately it's not so easy and may be blocked by https://github.com/facebook/docusaurus/pull/9619

---

As a workaround, `parseFrontMatter` may let you run some custom async code and provide an `image` blog front matter. But your code should rather be fast (cached on re-runs).

- https://docusaurus.io/blog/releases/3.1#parsefrontmatter-hook
- https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-blog#image

> I wanted to swizzle BlogPost, but Docusaurus says it's unsafe.

That's perfectly fine to swizzle it, we just don't guarantee semver retrocompatibility on it.
